misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/MatchResourcewithOperationId.java

```diff
@@ -0,0 +1,534 @@
+                List<String> servicePathOperations = openAPIPathSummary.getAvailableOperations();
+                if (resourcePath.equals(servicePath)) {
+                    isExit = true;
+                    if ((!servicePathOperations.isEmpty()) && (!resourcePathMethods.isEmpty())) {
+                        for (Map.Entry<String, ResourceMethod> entry : resourcePathMethods.entrySet()) {
+                            Boolean isMethodExit = false;
+                            for (String operation : servicePathOperations) {
+                                if (entry.getKey().equals(operation)) {
+                                    isMethodExit = true;
+                                    break;
+                                }
+                            }
+                            if (!isMethodExit) {
+                                ResourceValidationError resourceValidationError =
+                                        new ResourceValidationError(entry.getValue().getMethodPosition(),
+                                                entry.getKey(), resourcePath);
+                                resourceValidationErrorList.add(resourceValidationError);
+                            }
+                        }
+                    }
+                    break;
+                }
+            }
+            if (!isExit) {
+                ResourceValidationError resourceValidationError =
+                        new ResourceValidationError(resourcePathSummary.getPathPosition(), null, resourcePath);
+                resourceValidationErrorList.add(resourceValidationError);
+            }
+        }
+        return resourceValidationErrorList;
+    }
+
+    /**
+     * Checking the documented services are available at the resource file.
+     * @param openAPISummaries  openApi contract object
+     * @param serviceNode       resource file service
+     * @return                  validation error list type with OpenAPIServiceValidationError
+     */
+    public static List<OpenapiServiceValidationError> checkServiceAvailable(List<OpenAPIPathSummary> openAPISummaries,
+                                                                            ServiceNode serviceNode) {
+        List<OpenapiServiceValidationError> validationErrors = new ArrayList<>();
+        List<ResourcePathSummary> resourcePathSummaries = summarizeResources(serviceNode);
+//        check the contract paths are available at the resource
+        for (OpenAPIPathSummary openAPIPathSummary: openAPISummaries) {
+            Boolean isServiceExit = false;
+            for (ResourcePathSummary resourcePathSummary: resourcePathSummaries) {
+                if (openAPIPathSummary.getPath().equals(resourcePathSummary.getPath())) {
+                    isServiceExit = true;
+//                    check whether documented operations are available at resource file
+                    if ((!openAPIPathSummary.getAvailableOperations().isEmpty())) {
```

Comment:
Can we remove that extra bracket?
```suggestion
                    if (!openAPIPathSummary.getAvailableOperations().isEmpty()) {
```